avocado/core/output.py

```diff
@@ -709,11 +709,13 @@ def log_plugin_failures(failures):
                                   'skip_broken_plugin_notification',
                                   list, [])
     for failure in failures:
-        if failure[0].module_name in silenced:
+        plugin = getattr(failure[0], 'module', None)
```

Comment:
is it a little clearer like below:
   try:
         plugin = getattr(failure[0], 'module')
    except AttributeError as error:
         plugin =  getattr(failure[0], 'module_name')

Reply:
That's a good point. But there might be a better way than managing code logic by using exception catching method. I have thought about it myself too, and due to the reason I just stated, I have chosen this way. 
But, I'm open to all suggestions, that's how I can learn :grin: . Do you mind if I wait for some other suggestions and then make the decision? Thank you for reviewing.

Reply:
Wouldn't it make more sense to simply pin stevedore to a version before this API change/breakage?  As an added point, starting on version 71.0, stevedore is not needed anymore (that means the next LTS won't needed it as well).

Reply:
I believe it could. Given that "starting on version 71.0, stevedore is not needed anymore", which I'm just aware of, that seems a better way. Thank you for the suggestion.

Reply:
Any basic validation result available?

Reply:
```
(avo) [root@hp avocado]# pip show stevedore
Name: stevedore
Version: 2.0.1
Summary: Manage dynamic plugins for Python applications
Home-page: https://docs.openstack.org/stevedore/latest/
Author: OpenStack
Author-email: openstack-discuss@lists.openstack.org
License: UNKNOWN
Location: /home/chloe/venv/avo/lib/python3.6/site-packages/stevedore-2.0.1-py3.6.egg
Requires: pbr
You are using pip version 9.0.1, however version 20.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
(avo) [root@hp avocado]# avocado
Failed to load plugin from module "avocado_vt.plugins.vt": ImportError("Bootstrap missing. Execute 'avocado vt-bootstrap' or disable this plugin to get rid of this message",) :
  File "/home/chloe/venv/avo/lib/python3.6/site-packages/stevedore-2.0.1-py3.6.egg/stevedore/extension.py", line 195, in _load_plugins
    verify_requirements,
  File "/home/chloe/venv/avo/lib/python3.6/site-packages/stevedore-2.0.1-py3.6.egg/stevedore/enabled.py", line 79, in _load_one_plugin
    verify_requirements,
  File "/home/chloe/venv/avo/lib/python3.6/site-packages/stevedore-2.0.1-py3.6.egg/stevedore/extension.py", line 223, in _load_one_plugin
    plugin = ep.resolve()
  File "/home/chloe/venv/avo/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2297, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/chloe/codes/avocado-vt/avocado_vt/plugins/vt.py", line 41, in <module>
    raise ImportError("Bootstrap missing. "

Failed to load plugin from module "avocado_vt.plugins.vt_list": ImportError("Bootstrap missing. Execute 'avocado vt-bootstrap' or disable this plugin to get rid of this message",) :
  File "/home/chloe/venv/avo/lib/python3.6/site-packages/stevedore-2.0.1-py3.6.egg/stevedore/extension.py", line 195, in _load_plugins
    verify_requirements,
  File "/home/chloe/venv/avo/lib/python3.6/site-packages/stevedore-2.0.1-py3.6.egg/stevedore/enabled.py", line 79, in _load_one_plugin
    verify_requirements,
  File "/home/chloe/venv/avo/lib/python3.6/site-packages/stevedore-2.0.1-py3.6.egg/stevedore/extension.py", line 223, in _load_one_plugin
    plugin = ep.resolve()
  File "/home/chloe/venv/avo/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2297, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/chloe/codes/avocado-vt/avocado_vt/plugins/vt_list.py", line 26, in <module>
    from .vt import add_basic_vt_options, add_qemu_bin_vt_option
  File "/home/chloe/codes/avocado-vt/avocado_vt/plugins/vt.py", line 41, in <module>
    raise ImportError("Bootstrap missing. "

usage: avocado [-h] [-v] [--config [CONFIG_FILE]] [--show [STREAM[:LVL]]] [-s]
               {config,diff,distro,exec-path,list,multiplex,plugins,run,sysinfo,variants,vt-bootstrap}
               ...

Avocado Test Runner

optional arguments:
  -h, --help            show this help message and exit
  -v, --version         show program's version number and exit
  --config [CONFIG_FILE]
                        Use custom configuration from a file
  --show [STREAM[:LVL]]
                        List of comma separated builtin logs, or logging
                        streams optionally followed by LEVEL (DEBUG,INFO,...).
                        Builtin streams are: "app": application output;
                        "test": test output; "debug": tracebacks and other
                        debugging info; "remote": fabric/paramiko debug;
                        "early": early logging of other streams, including
                        test (very verbose); "all": all builtin streams;
                        "none": disables regular output (leaving only errors
                        enabled). By default: 'app'
  -s, --silent          disables regular output (leaving only errors enabled)

subcommands:
  valid subcommands

  {config,diff,distro,exec-path,list,multiplex,plugins,run,sysinfo,variants,vt-bootstrap}
                        subcommand help
    config              Shows avocado config keys
    diff                Shows the difference between 2 jobs.
    distro              Shows detected Linux distribution
    exec-path           Returns path to avocado bash libraries and exits.
    list                List available tests
    multiplex           Tool to analyze and visualize test variants and params
    plugins             Displays plugin information
    run                 Runs one or more tests (native test, test alias,
                        binary or script)
    sysinfo             Collect system information
    variants            Tool to analyze and visualize test variants and params
    vt-bootstrap        Avocado VT - implements the 'vt-bootstrap' subcommand

avocado: error: the following arguments are required: subcommand
```

As it shows, when stevedore version is 2.0.1, avocado is working well, without crashing.

Reply:
```
(avo) [root@hp avocado]# pip show stevedore
Name: stevedore
Version: 3.1.0
Summary: Manage dynamic plugins for Python applications
Home-page: https://docs.openstack.org/stevedore/latest/
Author: OpenStack
Author-email: openstack-discuss@lists.openstack.org
License: UNKNOWN
Location: /home/chloe/venv/avo/lib/python3.6/site-packages
Requires: importlib-metadata, pbr
You are using pip version 9.0.1, however version 20.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
(avo) [root@hp avocado]# avocado
Avocado crashed unexpectedly: 'EntryPoint' object has no attribute 'module_name'
You can find details in /var/lib/avocado/data/crashes/avocado-traceback-2020-07-14_12:21:23-dkfxh9sk.log

```

And after upgraded to version 3.1(used to be 3.0), avocado crashes.